src/anders_mathiesen_ex/ex03_project/test_sorting.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Anders Mathiesen'
+__email__ = 'andermat@nmbu.no'
+
+
+import random
+
+
+def sort_list(list_data):
+    new_data = list(list_data)
+    for number in range(len(list_data)):
+        if number == len(list_data) - 1:
+            break
+        first_number = new_data[number]
+        second_number = new_data[number + 1]
+        if first_number > second_number:
+            new_data[number] = second_number
+            new_data[number + 1] = first_number
+    return new_data
+
+
+def bubble_sort(random_list):
+    for _ in range(len(random_list)):
+        random_list = sort_list(random_list)
+    return list(tuple(random_list))
+
+
+def test_empty():
+    """Test that the sorting function works for empty list"""
+    empty_list = []
+    sorted_list = bubble_sort(empty_list)
+    assert sorted_list == empty_list
+
+
+def test_single():
+    """Test that the sorting function works for single-element list"""
+    single_list = [1]
+    sorted_list = bubble_sort(single_list)
+    assert sorted_list == single_list
+
+
+def test_sorted_is_not_original():
+    """
+    Test that the sorting function returns a new object.
+
+    Consider
+
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
+
+    Now sorted_data shall be a different object than data,
+    not just another name for the same object.
+    """
+    single_list = [3, 2, 1]
+    sorted_list = bubble_sort(single_list)
+    assert sorted_list != single_list
```

Comment:
here you should use `is not` as it compares the list objects themselves and not their values such as `!=`